File: serializers/gson/src/main/java/at/stefangeyer/challonge/serializer/gson/adapter/ParticipantWrapperListWrapperAdapter.java
package at.stefangeyer.challonge.serializer.gson.adapter;

import at.stefangeyer.challonge.model.wrapper.ParticipantWrapper;
import at.stefangeyer.challonge.model.wrapper.ParticipantWrapperListWrapper;
import com.google.gson.*;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class ParticipantWrapperListWrapperAdapter implements JsonDeserializer<ParticipantWrapperListWrapper> {

    @Override
    public ParticipantWrapperListWrapper deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        JsonArray jsonArray = json.getAsJsonArray();
        List<ParticipantWrapper> list = new ArrayList<>();

        for (JsonElement e : jsonArray) {
            ParticipantWrapper wrapper = context.deserialize(e, ParticipantWrapper.class);
            list.add(wrapper);
        }

        return new ParticipantWrapperListWrapper(list);
    }
}

File: src/main/kotlin/at/stefangeyer/challonge/serializer/gson/adapter/ParticipantWrapperListWrapperAdapter.kt
package at.stefangeyer.challonge.serializer.gson.adapter

import at.stefangeyer.challonge.model.wrapper.ParticipantWrapper
import at.stefangeyer.challonge.model.wrapper.ParticipantWrapperListWrapper
import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import java.lang.reflect.Type

class ParticipantWrapperListWrapperAdapter : JsonDeserializer<ParticipantWrapperListWrapper> {
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): ParticipantWrapperListWrapper {
        val jsonArray = json.asJsonArray
        val list = mutableListOf<ParticipantWrapper>()

        for (e in jsonArray) {
            val wrapper = context.deserialize<ParticipantWrapper>(e, ParticipantWrapper::class.java)
            list.add(wrapper)
        }

        return ParticipantWrapperListWrapper(list)
    }
}